tests/test_functional.py

```diff
@@ -123,3 +124,66 @@ def test_sees_error_message_if_user_already_registered(self, user, testapp):
         res = form.submit()
         # sees error
         assert 'Username already registered' in res
+
+
+class TestPlayer:
+    """Players."""
+
+    def test_list_players(self, testapp, players):
+        """Check that we can list players."""
+        res = testapp.get(url_for('dashboard.players'))
+        assert res.status_int == 200
+
+        found_players = []
+        for row in res.html.find('table').find('tbody').find_all('tr'):
+            found_players.append([col.text for col in row.find_all('td')])
+        assert len(players) == 2
+
+
+class TestGame:
+    """Games."""
+
+    def test_list_games(self, testapp, db):
+        """Check that we can list games."""
+        first_game = GameFactory(winner=Color.white, handicap=3, komi=0)
+        second_game = GameFactory(winner=Color.black, handicap=0, komi=7)
+        db.session.commit()
+
+        res = testapp.get(url_for('dashboard.list_games'))
+        assert res.status_int == 200
+
+        games = []
+        for row in res.html.find('table').find('tbody').find_all('tr'):
+            games.append([col.text for col in row.find_all('td')])
+        assert len(games) == 2
+        assert games[0] == [str(first_game.white.aga_id),
+                            str(first_game.black.aga_id),
+                            first_game.winner.name, str(first_game.handicap),
+                            str(first_game.komi)]
+        assert games[1] == [str(second_game.white.aga_id),
+                            str(second_game.black.aga_id),
+                            second_game.winner.name, str(second_game.handicap),
+                            str(second_game.komi)]
+
+    @pytest.mark.parametrize('winner', ['white'])
```

Comment:
<3 `py.test` so much. It's actually a bit problematic; I initially had way too many params here and the geometric growth in the test suite was absurd. I don't know re db, but for tests it's using SQLite so it's pretty fast in any case